app/src/main/java/com/zulip/android/networking/HTTPRequest.java

```diff
@@ -1,141 +1,115 @@
 package com.zulip.android.networking;
 
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpStatus;
-import org.apache.http.NameValuePair;
-import org.apache.http.StatusLine;
-import org.apache.http.client.HttpResponseException;
-import org.apache.http.client.entity.UrlEncodedFormEntity;
-import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.client.methods.HttpPut;
-import org.apache.http.client.methods.HttpRequestBase;
-import org.apache.http.client.utils.URLEncodedUtils;
-import org.apache.http.message.BasicNameValuePair;
-import org.apache.http.params.HttpConnectionParams;
-
-import android.net.http.AndroidHttpClient;
 import android.os.AsyncTask;
 import android.util.Base64;
 import android.util.Log;
 
+import com.zulip.android.R;
 import com.zulip.android.ZulipApp;
+import com.zulip.android.util.ZLog;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 
 /**
  * Simplified HTTP request API
  */
 public class HTTPRequest {
     ZulipApp app;
-    List<NameValuePair> nameValuePairs;
-    HttpRequestBase request;
-    HttpResponse response;
     volatile boolean aborting = false;
+    HashMap<String, String> properties;
+    OkHttpClient okHttpClient;
+    Response response = null;
+    Request request;
 
     public HTTPRequest(ZulipApp app) {
-        nameValuePairs = new ArrayList<NameValuePair>();
+        properties = new HashMap<>();
         this.app = app;
+        okHttpClient = new OkHttpClient.Builder().readTimeout(60, TimeUnit.SECONDS).build();
     }
 
+
     public void setProperty(String key, String value) {
-        this.nameValuePairs.add(new BasicNameValuePair(key, value));
+        properties.put(key, value);
     }
 
     public void clearProperties() {
-        this.nameValuePairs.clear();
+        properties.clear();
     }
 
     public void abort() {
         aborting = true;
-        if (request != null) {
+        if (response != null) {
             (new AsyncTask<Void, Void, Void>() {
                 @Override
                 protected Void doInBackground(Void... voids) {
-                    request.abort();
+                    response.body().close();
                     return null;
                 }
             }).execute();
         }
     }
 
-    // Java doesn't support a request body on DELETE requests, but RFC2616 does
-    // not prohibit it.
-    class HttpDeleteWithReq extends HttpPost {
```

Comment:
Here the Delete code extends from POST, but in your updated version, it appears to not support this behavior?